.buildkite/steps/tests.sh

```diff
@@ -6,8 +6,16 @@ echo arch is "$(uname -m)"
 
 go install gotest.tools/gotestsum@v1.8.0
 
+go install github.com/buildkite/test-engine-client@v1.5.0-rc.3
+
 echo '+++ Running tests'
-gotestsum --junitfile "junit-${BUILDKITE_JOB_ID}.xml" -- -count=1 -coverprofile=cover.out -failfast "$@" ./...
+export BUILDKITE_TEST_ENGINE_SUITE_SLUG=buildkite-agent
+export BUILDKITE_TEST_ENGINE_TEST_RUNNER=gotest
+export BUILDKITE_TEST_ENGINE_RESULT_PATH="junit-${BUILDKITE_JOB_ID}.xml"
+export BUILDKITE_TEST_ENGINE_RETRY_COUNT=1
+export BUILDKITE_TEST_ENGINE_TEST_CMD="gotestsum --junitfile={{resultPath}} -- -count=1 -coverprofile=cover.out -failfast $@ {{packages}}"
```

Comment:
I'm not sure how much we use the coverage report, but I _think_ this may clobber the coverage file if there is a retry.

https://go.dev/doc/build-cover#working
https://dustinspecker.com/posts/go-combined-unit-integration-code-coverage/#run-unit-tests-to-collect-coverage

A quick google shows that the newer binary coverage format can be merged and has some other niceties though, so that could be something we adopt?

Also, do we want `-failfast` if we're using bktec too - I assume we proably want to run the whole subset of the suite?

Reply:
I removed `-failfast`. 

For the coverage report change, I raised another PR: https://github.com/buildkite/agent/pull/3295 🙏🏿